src/authentification/components/AuthRegisterForm.tsx

```diff
@@ -30,27 +46,65 @@ export default function AuthRegisterForm() {
 
   return (
     <FormProvider {...form}>
-      <form onSubmit={form.handleSubmit(onSubmit)}>
-        <h2>Register</h2>
-        <label>Enter a nickname</label>
-        <input
-          {...form.register("name", {
-            required: true,
-            minLength: {
-              value: usernameMinLength,
-              message: `Nickname must be atleast ${usernameMinLength} characters long.`,
-            },
-            maxLength: {
-              value: usernameMaxLength,
-              message: `Nickname cannot be longer than ${usernameMaxLength} characters.`,
-            },
-          })}
-        />
-        <label>Enter your e-mail</label>
-        <EmailInput email="email" />
-        <label>Enter a password</label>
-        <PasswordInput password="password" />
-        <input type="submit" />
+      <form
+        id="register-form"
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="flex flex-col w-screen sm:w-[500px] bg-basic-white text-text-default-gray px-10 py-8 sm:shadow-md"
+      >
+        <h2 className="form-size-header font-bold">Sign Up</h2>
+        <span className="text-sm">Don't have an account? Create one now!</span>
+        <section className="flex flex-col justify-between h-[320px] my-5 pb-5 sm:pb-10">
+          <label>Nickname</label>
+          <InputField
+            name="name"
+            placeholder="Enter a nickname..."
+            validation={{
+              required: true,
+              minLength: {
+                value: usernameMinLength,
+                message: `Nickname must be atleast ${usernameMinLength} characters long.`,
+              },
+              maxLength: {
+                value: usernameMaxLength,
+                message: `Nickname cannot be longer than ${usernameMaxLength} characters.`,
+              },
+            }}
+          />
+          <label>E-mail address</label>
+          <InputField
+            name="email"
+            placeholder="Enter your email..."
+            validation={{ required: true }}
+          />
+          <label>Password</label>
+          <InputField
```

Comment:
Same as in the Login form. As Stjepan said, when you pass the rest of the props to the InputField component you could then pass `type` from password and email

Reply:
Added type for all InputFields.